### PR TITLE
don't write null to stream if response is empty

### PR DESCRIPTION
**Pull request by tungfinblox**

### CODE REVIEW

**Comment by dunglas** [comment]
Could you add a test please? Also, isn't it possible to prevent the call to `isEmpty()` by just checking the return value of `getContent()`?

**Comment by tungfinblox** [comment]
> 

Hi. Yes I've just did that. 
My old approach based on `Dingo\Api\Http\Response` and that one based on an old version of `\Symfony\Component\HttpFoundation\Response` which still allows `$content` to be null. And that how the bug happened.

Now current version of `\Symfony\Component\HttpFoundation\Response` enforce return type of `getContent` must be `string|false` so I've just updated the PR following that. And this fix will still work for the old version of `\Symfony\Component\HttpFoundation\Response`

File: Factory/PsrHttpFactory.php [hunk 1]
```diff
@@ -143,7 +143,7 @@ public function createResponse(Response $symfonyResponse)
                 $symfonyResponse->sendContent();
                 ob_end_clean();
             } else {
-                $stream->write($symfonyResponse->getContent());
+                $stream->write(is_string($symfonyResponse->getContent()) ? $symfonyResponse->getContent() : '');
```
Comment (dunglas) [inline Factory/PsrHttpFactory.php:146]: Maybe just this?

```suggestion
                $stream->write((string) $symfonyResponse->getContent());
```

Comment (tungfinblox) [reply Factory/PsrHttpFactory.php:146]: Yeah I thought about this but I'm not sure what`false` will be when it is converted to string.

Comment (derrabus) [reply Factory/PsrHttpFactory.php:146]: https://3v4l.org/Nnba7

Comment (tungfinblox) [reply Factory/PsrHttpFactory.php]: cool. Thank you. I've just fixed it.

**Review by dunglas — APPROVED** [review]

**Comment by tungfinblox** [comment]
Hi. Looks like one of the checks is stucked?

**Comment by derrabus** [comment]
I'm not sure if this if the correct fix. If `getContent()` returns false, this usually does not mean that there is no content. It means that the content is generated when sending the response.

**Comment by tungfinblox** [comment]
> I'm not sure if this if the correct fix. If `getContent()` returns false, this usually does not mean that there is no content. It means that the content is generated when sending the response.

Is that case handled in: `https://github.com/symfony/psr-http-message-bridge/blob/main/Factory/PsrHttpFactory.php#L136`? And yeah I started to think that this might not be necessary because I got the bug because of the behavior of `Dingo\Api\Http\Response` when it returns 304 status code and empty content. And this library might not be responsible for handling such case.

**Pull request closed without merging**